iceoryx_posh/source/popo/user_trigger.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2020 by Robert Bosch GmbH, Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/popo/user_trigger.hpp"
+#include "iceoryx_posh/popo/wait_set.hpp"
+
+namespace iox
+{
+namespace popo
+{
+cxx::expected<WaitSetError> UserTrigger::attachToWaitset(WaitSet& waitset,
+                                                         const uint64_t triggerId,
+                                                         const Trigger::Callback<UserTrigger> callback) noexcept
+{
+    m_trigger.reset();
```

Comment:
First resetting, then assigning a default, then creating a new and moving into this. Say what?

Reply:
Whoopsy ... and fixed